### PR TITLE
Command context updates

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
This adds support for parent context for commands in the CmdRef entries.

Here are two examples to help understand the code.

```yaml
---
_template: # _template holds common settings for all commands
  # Enable feature telemetry if disabled
  feature: telemetry
  # Common get syntax for TMS commands
  get_command: show run telemetry all
  # Parent configuration for TMS commands
  context:
    - telemetry

certificate:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: dict
  getval: certificate (?P<key>\S+) (?P<hostname>\S+)$
  setval: certificate {key} {hostname}
  default:
    key: ~
    hostname: ~

destination_profile_compression:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: str
  getval: use-compression (\S+)$
  setval: 'use-compression {0}'
  default: ~
  context: [telemetry, destination-profile]

destination_profile_source_interface:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: str
  getval: source-interface (\S+)$
  setval: 'source-interface {0}'
  default: ~
  context: [telemetry, destination-profile]

destination_profile_vrf:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: str
  getval: use-vrf (\S+)$
  setval: 'use-vrf {0}'
  default: ~
  context: [telemetry, destination-profile]
"""
```

In this example, the global context is used but is overwritten by the more specific property context.

```yaml
---
_template: # _template holds common settings for all commands
  # Enable feature telemetry if disabled
  feature: telemetry
  # Common get syntax for TMS commands
  get_command: show run telemetry all
  # Parent configuration for TMS commands
  context:
    - telemetry

identifier:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: int
  getval: destination-group (\S+)$
  setval: 'destination-group {0}'
  default: ~

destination:
  _exclude: ['N3K', 'N5K', 'N6k', 'N7k']
  kind: dict
  getval: ip address (?P<ip>\S+) port (?P<port>\S+) protocol (?P<protocol>\S+) encoding (?P<encoding>\S+)$
  setval: ip address {ip} port {port} protocol {protocol} encoding {encoding}
  default:
    ip: ~
    port: ~
    protocol: ~
    encoding: ~
  context: ['telemetry', 'setval::identifier']
```

In this example, the context for destination is `['telemetry', 'destination-group {val}']`.  The `setval::identifier` is a placeholder to use the `setval` from the `identifier` property.

**Note:** This does not yet have support for `sub_context`.  This will be part of a different PR.

Unit tests for BFD pass.

### CODE REVIEW

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import re
 import sys
 import yaml
+from collections import OrderedDict
```
Comment (chrisvanheuveln) [inline lib/ansible/module_utils/network/nxos/nxos.py:36]: This *might* be a problem as `OrderedDict` originated in 2.7, but one of the shippable test versions is 2.6.

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 2]
```diff
@@ -969,24 +970,38 @@ def get_proposed(self):
...
-                            proposed.append(ctx_cmd)
-                    elif isinstance(context, str):
-                        proposed.append(context)
-
-                proposed.append(cmd)
-
-        return proposed
+                # Add processed command to cmd_ref object
+                ref[k]['setcmd'] = cmd
+
```
Comment (chrisvanheuveln) [inline lib/ansible/module_utils/network/nxos/nxos.py:1008]: I'd prefer to have the new code below as a separate method; e.g. `proposed_context()`.

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 3]
```diff
@@ -969,24 +970,38 @@ def get_proposed(self):
...
-                        for ctx_cmd in context:
-                            proposed.append(ctx_cmd)
-                    elif isinstance(context, str):
-                        proposed.append(context)
-
-                proposed.append(cmd)
-
-        return proposed
+                # Add processed command to cmd_ref object
+                ref[k]['setcmd'] = cmd
```
Comment (chrisvanheuveln) [inline lib/ansible/module_utils/network/nxos/nxos.py]: Nit: I don't like calling this `setcmd` - it's a bit confusing with `setval`. Can you use `'proposed_cmd'` instead?

File: lib/ansible/module_utils/network/nxos/nxos.py (hunk 1 above)
Comment (mikewiebe) [reply lib/ansible/module_utils/network/nxos/nxos.py:36]: It won't be.. I already used this to fix a bug in this file several months back when building nxapi requests.

Reference: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/nxos/nxos.py#L304

Comment (chrisvanheuveln) [reply lib/ansible/module_utils/network/nxos/nxos.py:36]: cool

**Pull request merged**